### PR TITLE
Make signature for slicing compliant with spec

### PR DESCRIPTION
**Pull request by bettinaheim**

### CODE REVIEW

File: src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestCaching2.ll [hunk 1]
```diff
@@ -29,7 +29,7 @@ condContinue__2:                                  ; preds = %condFalse__2, %cond
   %7 = insertvalue %Range %6, i64 %2, 0
   %8 = insertvalue %Range %7, i64 2, 1
   %9 = insertvalue %Range %8, i64 %5, 2
-  %10 = call %Array* @__quantum__rt__array_slice_1d(%Array* %vals, %Range %9)
+  %10 = call %Array* @__quantum__rt__array_slice_1d(%Array* %vals, %Range %9, i1 true)
```
Comment (kuzminrobin) [inline src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestCaching2.ll:32]:
```suggestion
  %10 = call %Array* @__quantum__rt__array_slice_1d(%Array* %vals, %Range %9, i1 zeroext true)
```
(more [here](https://llvm.org/docs/LangRef.html#parameter-attributes) about the `zeroext`)

File: src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestRange.ll [hunk 2]
```diff
@@ -33,7 +33,7 @@ entry:
   store i64 14, i64* %18, align 4
   store i64 16, i64* %20, align 4
   call void @__quantum__rt__array_update_alias_count(%Array* %a, i32 1)
-  %b = call %Array* @__quantum__rt__array_slice_1d(%Array* %a, %Range %x)
+  %b = call %Array* @__quantum__rt__array_slice_1d(%Array* %a, %Range %x, i1 true)
```
Comment (kuzminrobin) [inline src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestRange.ll:36]:
```suggestion
  %b = call %Array* @__quantum__rt__array_slice_1d(%Array* %a, %Range %x, i1 zeroext true)
```

File: src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestUsing2.ll [hunk 3]
```diff
@@ -14,7 +14,7 @@ entry:
   %3 = insertvalue %Range %2, i64 0, 0
   %4 = insertvalue %Range %3, i64 2, 1
   %5 = insertvalue %Range %4, i64 %max, 2
-  %y = call %Array* @__quantum__rt__array_slice_1d(%Array* %b, %Range %5)
+  %y = call %Array* @__quantum__rt__array_slice_1d(%Array* %b, %Range %5, i1 true)
```
Comment (kuzminrobin) [inline src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestUsing2.ll:17]:
```suggestion
  %y = call %Array* @__quantum__rt__array_slice_1d(%Array* %b, %Range %5, i1 zeroext true)
```

**Review by kuzminrobin — APPROVED** [review]
Approved with some comments.

**Review by ricardo-espinoza — APPROVED** [review]

File: src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestCaching2.ll (hunk 1 above)
Comment (bettinaheim) [reply src/QsCompiler/Tests.Compiler/TestCases/QirTests/TestCaching2.ll:32]: this is autogenerated, and the change to do that across the board will be done separately (and not included in this release)

**Pull request merged**